unity-renderer/Assets/Scripts/MainScripts/DCL/Components/Avatar/AvatarShape.cs

```diff
@@ -47,12 +47,12 @@ protected override void OnEnable()
         {
             base.OnEnable();
 
-            if (disableFacialFeatureRoutine != null)
+            if (checkDistanceRoutine != null)
             {
-                StopCoroutine(disableFacialFeatureRoutine);
-                disableFacialFeatureRoutine = null;
+                StopCoroutine(checkDistanceRoutine);
+                checkDistanceRoutine = null;
             }
-            disableFacialFeatureRoutine = StartCoroutine(SetFacialFeaturesVisibleRoutine());
+            checkDistanceRoutine = StartCoroutine(CheckDistanceToPlayerRoutine());
```

Comment:
Please let's put this logic inside a specialized class, and try to contain it in `AvatarRenderer`. `AvatarShape` shouldn't be the place for putting performance related plugins. Yes, `AvatarRenderer` is in emergency mode, but even so its a better place to put this into.

Reply:
This shouldn't go in `AvatarRenderer`. `AvatarRenderer` is used by the AvatarEditor and the own player for the 3rd person Camera.

This feature (and the disabling of the facial features) is just related to how we render other players. IMO the capability of enabling/disabling SSAO and Facial Features should be in the AvatarRenderer (as it is in the implementation), but the decision should be external (in the AvatarShape in this case).